lib/cucumber/core/report/summary.rb

```diff
@@ -4,20 +4,22 @@ module Report
       class Summary
         attr_reader :test_cases, :test_steps
 
-        def initialize
+        def initialize(event_bus)
           @test_cases = Test::Result::Summary.new
           @test_steps = Test::Result::Summary.new
+          subscribe_to(event_bus)
         end
 
-        def after_test_case(test_case, result)
-          result.describe_to test_cases
-        end
-
-        def after_test_step(test_step, result)
-          result.describe_to test_steps
-        end
+        private
 
-        def method_missing(*)
+        def subscribe_to(events)
```

Comment:
I think this could be refactored too, like so:
```ruby
def subscribe_to(events)
  subscribe_to_event(events, :test_case_finished, test_cases)
  subscribe_to_event(events, :test_step_finished, test_steps)
  self
end

private

def subscribe_to_event(events, event, describe_to)
  events.on(event) do |e|
    event.result.describe_to describe_to
  end
end
```
Just maybe don't name the variables the same things I did, I suck at naming variables 😉